aten/src/ATen/native/CPUBlas.h

```diff
@@ -186,4 +193,364 @@ void copy(int64_t n, const float *x, int64_t incx, float *y, int64_t incy);
 void copy(int64_t n, const c10::complex<double> *x, int64_t incx, c10::complex<double> *y, int64_t incy);
 void copy(int64_t n, const c10::complex<float> *x, int64_t incx, c10::complex<float> *y, int64_t incy);
 
-}  // namespace at::native::cpublas
+#if AT_MKLDNN_ENABLED() && (defined(__x86_64__) || (defined(_M_X64) && !defined(_M_ARM64EC)))
+template <typename key_t, typename value_t>
+struct KernelCache  {
```

Comment:
Is there any advantage to having these classes in the header file? They seem to be private implementation details. In fact it should probably be moved to the mkldnn folder.

Reply:
Moved such codes into `CPUBlas.cpp` as brgemm are the basic ukernel like other `gemm` defined in `CPUBlas.cpp`.